errata_tool/build.py

```diff
@@ -0,0 +1,101 @@
+from errata_tool import ErrataConnector
+from errata_tool import Erratum
+
+
+class Build(ErrataConnector):
+    def __init__(self, nvr):
+        """
+        Find build information and advisories where the build is attached.
+
+        :param nvr: build NVR (name-version-release)
+
+        """
+        self.nvr = nvr
+        self.url = self._url + '/api/v1/build/%s' % self.nvr
+        self._released_errata = None
+        self._all_errata = []
+        self._signed_rpms = None
+        self._data = {}
+        self._files = []
+
+        self._fetch()
+
+    def _fetch(self):
+        """
+        Fetch build data from Errata Tool API and store them to properties
+        """
+        self._data = self._get(self.url)
+
+        if self._data['released_errata'] is not None:
+            self._released_errata = Erratum(errata_id=self._data[
+                'released_errata']['id'])
+
+        for errata_dict in self._data['all_errata']:
+            errata = Erratum(errata_id=errata_dict['id'])
+            self._all_errata.append(errata)
```

Comment:
This part takes a while to fetch the advisories.

`$ time errata-tool build get ceph-ansible-3.0.27-1.el7cp` says 1m23.411s

When I take out the `Erratum` constructor, it runs in under two seconds, like
```
diff --git a/errata_tool/build.py b/errata_tool/build.py
index 444ca60..d5500d5 100644
--- a/errata_tool/build.py
+++ b/errata_tool/build.py
@@ -27,11 +27,11 @@ class Build(ErrataConnector):
         self._data = self._get(self.url)
 
         if self._data['released_errata'] is not None:
-            self._released_errata = Erratum(errata_id=self._data[
-                'released_errata']['id'])
+            self._released_errata = self._data[
+                'released_errata']['id']
 
         for errata_dict in self._data['all_errata']:
-            errata = Erratum(errata_id=errata_dict['id'])
+            errata = errata_id=errata_dict['id']
             self._all_errata.append(errata)
```

(I had to modify `__str__` as well to just print the ID numbers)

What do you think? Callers would have to instantiate the `Erratum` classes themselves for now, but the default behavior could be a lot faster. 

I know we've got to resolve the performance of the `Erratum` class constructor long-term. If we wanted to leave open the possibility of fixing that and then providing real `Erratum` objects down the road here when that constructor is faster, we could name these properties `all_errata_ids` and `release_errata_id` today, so we have room to return real `all_errata` and `release_errata` later.

Reply:
Or maybe even just return the plain `self._data['all_errata']` list of dicts here... I had the same problem with `Release.advisories()`

Reply:
Ok I've talked myself into merging this as-is, and we can add the speed improvements later.